espflash/src/interface.rs

```diff
@@ -29,7 +36,7 @@ pub enum SerialConfigError {
 /// implemented.
 pub struct Interface {
     /// Hardware serial port used for communication
-    pub serial_port: Box<dyn SerialPort>,
+    pub serial_port: Port,
```

Comment:
As pointed by @jessebraham, this change would require bumping the SemVer version major, hence, the PR will be reviewed once we start working on v3.0.0